icu4c/source/i18n/myancal.cpp

```diff
@@ -0,0 +1,501 @@
+		{653, 2}, {656, 1}, {672, 1}, {729, 1}, {767, -1}};
+		long wte11[] = {-9999};
+		big_watat = era_big_watat;
+    wte = wte11;
+		uf = long(sizeof(era_big_watat) / sizeof(era_big_watat[0]));
+		uw = long(sizeof(wte11) / sizeof(wte11[0]));
+	}
+
+	i = bSearch2(myan_year, big_watat, uf);
+  if (i >= 0) WO += big_watat[i][1]; // full moon day offset exceptions
+
+  i = bSearch1(myan_year, wte, uw);
+  if (i >= 0) EW = 1; //correct watat exceptions
+}
+
+void MyanmarCalendar::cal_watat(int32_t myan_year, long& watat, long& full_moon_waso_2) const {//get data for respective era
+	double era, WO, NM, excess_days;
+  long EW;
+	GetMyConst(myan_year, era, WO, NM, EW); // get constants for the corresponding calendar era
+	double TA = (SOLAR_YEAR / 12 - LUNAR_MONTH) * (12 - NM); //threshold to adjust
+	ClockMath::floorDivide(SOLAR_YEAR * (myan_year + 3739), LUNAR_MONTH, excess_days); // excess days
+	if (excess_days < TA) {
+    excess_days += LUNAR_MONTH; //adjust excess days
+  }
+	full_moon_waso_2 = long(SOLAR_YEAR * myan_year + MYANMAR_EPOCH - excess_days + 4.5 * LUNAR_MONTH + WO);
+  //full moon day of 2nd Waso
+	double TW = 0;
+  watat = 0;//find watat
+	if (era >= 2) {//if 2nd era or later find watat based on excess days
+		TW = LUNAR_MONTH - (SOLAR_YEAR / 12 - LUNAR_MONTH) * NM;
+		if (excess_days >= TW) {
+      watat = 1;
+    }
+	}
+	else {//if 1st era,find watat by 19 years metonic cycle
+	//Myanmar year is divided by 19 and there is intercalary month
+	//if the remainder is 2,5,7,10,13,15,18
+	//https://github.com/kanasimi/CeJS/blob/master/data/date/calendar.js#L2330
+		watat = (myan_year * 7 + 2) % 19;
+    if (watat < 0) {
+      watat += 19;
+    }
+		watat = ClockMath::floorDivide(watat, 12);
+	}
+	watat^=EW;//correct watat exceptions
+}
+
+// Myanmar does not change clocks for Daylight Time
+UBool
+MyanmarCalendar::inDaylightTime(UErrorCode& status) const
+{
+    // copied from GregorianCalendar
+    if (U_FAILURE(status) || !getTimeZone().useDaylightTime())
+        return FALSE;
+
+    // Force an update of the state of the Calendar.
+    ((MyanmarCalendar*)this)->complete(status); // cast away const
+
+    return (UBool)(U_SUCCESS(status) ? (internalGet(UCAL_DST_OFFSET) != 0) : FALSE);
+}
+
+// default century
+
+static UDate           gSystemDefaultCenturyStart       = DBL_MIN;
+static int32_t         gSystemDefaultCenturyStartYear   = -1;
+static icu::UInitOnce  gSystemDefaultCenturyInit        = U_INITONCE_INITIALIZER;
+
+UBool MyanmarCalendar::haveDefaultCentury() const
+{
+    return TRUE;
+}
+
+static void U_CALLCONV initializeSystemDefaultCentury() {
+    // initialize systemDefaultCentury and systemDefaultCenturyYear based
+    // on the current time.  They'll be set to 80 years before
+    // the current time.
+    UErrorCode status = U_ZERO_ERROR;
+    MyanmarCalendar calendar(Locale("@calendar=myanmar"),status);
+   if (U_SUCCESS(status))
+   {
+      printf("success\n");
```

Comment:
printf… put it  behind `#if defined (U_DEBUG_CAL)  … #endif` and mention the file and line, see https://github.com/unicode-org/icu/blob/85c7e61b3e788be448a4ea96eb8861c1a4003d94/icu4c/source/i18n/gregocal.cpp#L377

Reply:
(will also need `#if defined (U_DEBUG_CAL)  …   #include <stdio.h> … #endif` earlier in the source file)